helix-term/src/commands.rs

```diff
@@ -2505,17 +2505,19 @@ async fn make_format_callback(
     let format = format.await?;
     let call: job::Callback = Box::new(move |editor, _compositor| {
         let view_id = view!(editor).id;
-        if let Some(doc) = editor.document_mut(doc_id) {
-            if doc.version() == doc_version {
-                doc.apply(&format, view_id);
-                doc.append_changes_to_history(view_id);
-                doc.detect_indent_and_line_ending();
-                if let Modified::SetUnmodified = modified {
-                    doc.reset_modified();
-                }
-            } else {
-                log::info!("discarded formatting changes because the document changed");
+        let scrolloff = editor.config().scrolloff;
+        let doc = doc_mut!(editor, &doc_id);
```

Comment:
This macro has an `unwrap` that might cause a panic if the document has been closed after the formatting callback completes. Instead we should return early if we see that the doc is `None`:

```suggestion
        let doc = match editor.documents.get_mut(&doc_id) {
            Some(doc) => doc,
            None => return,
        };
```